spec/ndx-patterned-ogen.extensions.yaml

```diff
@@ -1,8 +1,165 @@
+          mean [um] ± s.d [um].
+      - name: axial_point_spread_function
+        dtype: text
+        doc:
+          Estimated axial spatial profile or point spread function, expressed as mean
+          [um]± s.d [um].
+  - neurodata_type_def: PatternedOptogeneticStimulusSite
+    neurodata_type_inc: OptogeneticStimulusSite
+    doc: Patterned optogenetic stimulus site.
+    attributes:
+      - name: effector
+        dtype: text
+        doc: Light-activated effector protein expressed by the targeted cell (e.g., ChR2).
+    links:
+      - name: spatial_light_modulator
+        target_type: SpatialLightModulator
+        doc: SpatialLightModulator used to generate holographic pattern.
+      - name: light_source
+        target_type: LightSource
+        doc: LightSource used to apply photostimulation.
+  - neurodata_type_def: SpatialLightModulator
+    neurodata_type_inc: Device
+    doc: Spatial light modulator used in the experiment.
+    attributes:
+      - name: model
+        dtype: text
+        doc: Model of SpatialLightModulator.
+        required: false
+      - name: size
+        dtype: numeric
+        doc:
+          Resolution of SpatialLightModulator (in pixels), formatted as [width, height]
+          or [width, height, depth].
+        required: false
+  - neurodata_type_def: LightSource
+    neurodata_type_inc: Device
+    doc: Light source used in the experiment.
+    attributes:
+      - name: model
+        dtype: text
+        doc: Model of LightSource.
+        required: false
+      - name: stimulation_wavelength
+        dtype: numeric
+        doc: Excitation wavelength of stimulation light (nanometers).
+      - name: peak_power
+        dtype: numeric
+        doc: Incident power of stimulation device (in Watts).
+        required: false
+      - name: peak_pulse_energy
+        dtype: numeric
+        doc: If device is pulsed laser, pulse energy (in Joules).
+        required: false
+      - name: intensity
+        dtype: numeric
+        doc: Intensity of the excitation in W/m^2, if known.
+        required: false
+      - name: exposure_time
+        dtype: numeric
+        doc: exposure time of the sample
+        required: false
+      - name: pulse_rate
+        dtype: numeric
+        doc: If device is pulsed laser, pulse rate (in Hz) used for stimulation.
+        required: false
+  - neurodata_type_def: OptogeneticStimulusTarget
+    neurodata_type_inc: LabMetaData
+    doc: Container to store the targated rois in a photostimulation experiment.
+    datasets:
+      - name: segmented_rois
+        neurodata_type_inc: DynamicTableRegion
+        doc: A table region corresponding to the ROIs that were targeted and stimulated.
+        required: false
+    attributes:   
+      - name: targeted_rois
+        dtype: numeric
+        doc: Targeted ROIs defined by their centroids, in pixel ([x1, y1], [x2, y2], ...) or voxel ([x1, y1, z1], [x2, y2, z2], ...).
```

Comment:
Similar to argument about 2D/3D separation, this field would be greatly simplified by offloading to a secondary table (where this would just be a `DynamicTableRegion` reference and dimensional complexities belong to the target object)

Reply:
Agree!